docs/kb/semgrep-code/semgrep-scan-troubleshooting.md

```diff
@@ -37,6 +38,14 @@ After addressing files to ignore:
   * Identify the slowest files from the list. You may find that you can add some of those files to your ignore list as well.
   * Identify the slowest rules from the list. You may find that some of them don't apply to your codebase and can be skipped.
 
+ ### Adjusting timeouts
+
+ Semgrep also has several timeout settings that affect scan duration, and can be adjusted to optimize scan behavior:
+
+ * `--timeout`: Similar to `--max-memory`, `--timeout` affects the behavior of the scan for a single rule-file combination. It defaults to 5 seconds. Typical values range from 3 seconds (favors faster scans, but more timeouts) to 30 seconds (slower scans, fewer timeouts). 
+ * `--timeout-threshold`: The number of times to try to scan a single rule-file combination, if it times out due to the `--timeout` limit. It defaults to 3. Decreasing the value may speed up scans but cause more timeouts.
```

Comment:
I'd be fine with "attempts made" but just "attempts" sounds weird to me. Anyway, if we need to update the single rule-file thing, this will need another update so not adding it yet.

Reply:
attempts made is good, too!